packages/private/edtr-io/__tests__/deserialize/rows-to-solution-steps.ts

```diff
@@ -39,19 +50,252 @@ test('three column layout', () => {
   ]
   expect(rowsToSolutionSteps(rows)).toEqual([
-    { type: 'step', isHalf: false, content: { plugin: 'text', state: {} } },
+    {
+      type: 'step',
+      isHalf: false,
+      content: { plugin: 'rows', state: [{ plugin: 'text', state: {} }] }
+    },
     {
       type: 'step',
       isHalf: false,
       content: {
-        plugin: 'layout',
+        plugin: 'rows',
         state: [
-          { child: { plugin: 'text', state: {} }, width: 1 },
-          { child: { plugin: 'text', state: {} }, width: 1 },
-          { child: { plugin: 'text', state: {} }, width: 1 }
+          {
+            plugin: 'layout',
+            state: [
+              { child: { plugin: 'text', state: {} }, width: 1 },
+              { child: { plugin: 'text', state: {} }, width: 1 },
+              { child: { plugin: 'text', state: {} }, width: 1 }
+            ]
+          }
         ]
       }
     },
-    { type: 'step', isHalf: false, content: { plugin: 'text', state: {} } }
+    {
+      type: 'step',
+      isHalf: false,
+      content: { plugin: 'rows', state: [{ plugin: 'text', state: {} }] }
+    }
   ])
 })
+
+test('migrate new state: introduction', () => {
+  const oldState: SolutionPlugin = {
+    plugin: 'solution',
+    state: [
+      {
+        plugin: 'solutionSteps',
+        state: {
+          introduction: { plugin: 'text', state: {} },
+          additionals: undefined,
+          strategy: undefined,
+          solutionSteps: [
+            {
+              isHalf: false,
+              type: 'step',
+              content: {
+                plugin: 'rows',
+                state: [{ plugin: 'text', state: {} }]
+              }
+            }
+          ]
+        }
+      }
+    ]
+  }
+
+  const newState: SolutionPlugin = {
+    plugin: 'solution',
+    state: [
+      {
+        plugin: 'solutionSteps',
+        state: {
+          introduction: {
+            plugin: 'rows',
+            state: [{ plugin: 'text', state: {} }]
+          },
+          additionals: undefined,
+          strategy: undefined,
+          solutionSteps: [
+            {
+              isHalf: false,
+              type: 'step',
+              content: {
+                plugin: 'rows',
+                state: [{ plugin: 'text', state: {} }]
+              }
+            }
+          ]
+        }
+      }
+    ]
+  }
+
+  expect(migrateSolutionStepsState(oldState)).toEqual(newState)
+})
+
+test('migrate new state: step without rows', () => {
+  const oldState: SolutionPlugin = {
+    plugin: 'solution',
+    state: [
+      {
+        plugin: 'solutionSteps',
+        state: {
+          introduction: {
+            plugin: 'rows',
+            state: [{ plugin: 'text', state: {} }]
+          },
+          additionals: undefined,
+          strategy: undefined,
+          solutionSteps: [
+            {
+              isHalf: false,
+              type: 'step',
+              content: { plugin: 'image', state: {} }
```

Comment:
Is this state even possible? The content of a solution step should already be a `rows` plugin, see https://github.com/serlo/serlo.org/blob/master/packages/private/edtr-io/src/plugins/solution-steps/index.ts#L19

Reply:
it is and it happened. When converting a normal one column row with deserialize, it took the plugin without wrapping it in a rows plugin